kiwi/bootloader/config/systemd_boot.py

```diff
@@ -0,0 +1,110 @@
+# Copyright (c) 2022 Marcus Schäfer
+#
+# This file is part of kiwi.
+#
+# kiwi is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# kiwi is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with kiwi.  If not, see <http://www.gnu.org/licenses/>
+#
+import os
+from string import Template
+from typing import Dict
+
+# project
+from kiwi.path import Path
+from kiwi.bootloader.template.systemd_boot import BootLoaderTemplateSystemdBoot
+from kiwi.bootloader.config.bootloader_spec_base import BootLoaderSpecBase
+from kiwi.command import Command
+
+from kiwi.exceptions import (
+    KiwiTemplateError,
+    KiwiBootLoaderTargetError
+)
+
+
+class BootLoaderSystemdBoot(BootLoaderSpecBase):
+    def create_loader_image(self, target: str) -> None:
+        self.efi_boot_path = self.create_efi_path()
+
+    def setup_loader(self, target: str) -> None:
+        if target != self.target.disk:
+            raise KiwiBootLoaderTargetError(
+                'systemd-boot is only supported with the EFI disk image target'
+            )
+        boot_options = self.custom_args['boot_options']
+        self._mount_system(
+            boot_options.get('root_device'),
+            boot_options.get('boot_device'),
+            boot_options.get('efi_device'),
+            boot_options.get('system_volumes')
+        )
+        BootLoaderSystemdBoot._write_config_file(
+            BootLoaderTemplateSystemdBoot().get_loader_template(),
+            self.root_mount.mountpoint + '/boot/efi/loader/loader.conf',
+            self._get_template_parameters()
+        )
+        Command.run(
+            ['chroot', self.root_mount.mountpoint, 'systemd-machine-id-setup']
+        )
```

Comment:
So in newer versions of systemd, we reworked systemd-boot a bit to not require a machine ID to be present in the final image which you want to avoid since each instance of the image should have its own machine ID (which is usually generated on first boot). So depending on which version of bootctl you want to support, you might want to make use of the new --entry-token option. If you want to support older versions, you should definitely remove the machine ID again after installing systemd-boot so you don't end up shipping the image with a machine ID.

Reply:
yep thanks for the clarification. I did all tests with Fedora (37) now as it has proper systemd-boot integration and made use of the --entry-token option which makes a lot sense to me